### ISSUE TITLE
gnt-node evacuate on a downed, offline node is behaving oddly

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1007]
Originally reported of Google Code with ID 951.

```
gnt-cluster (ganeti v2.11.5) 2.11.5

Software version: 2.11.5
Internode protocol: 2110000
Configuration format: 2110000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.11.5

hspace (ganeti) version v2.11.5
compiled with ghc 7.4
running on linux x86_64

Debian 7.4

Problem:

3 node cluster, (c0,c1,c2).There is 1 active DRBD primary instance on c1

Then c1 is shutdown:

root@c0:~# gnt-node list
Node           DTotal DFree MTotal MNode MFree Pinst Sinst
c0.ws.nsrc.org  30.0G 27.9G   1.5G  238M  1.3G     0     1
c1.ws.nsrc.org      ?     ?      ?     ?     ?     1     0
c2.ws.nsrc.org  30.0G 30.0G   1.5G   87M  1.4G     0     0

I offline c1:

root@c0:~# gnt-node modify --offline=yes c1
Thu Sep 18 20:11:26 2014  - WARNING: Communication failure to node 93eef5d9-6b33-4ce6-8f88-5353a093cb73: Error 7: Failed connect to 10.10.0.234:1811; No route to host
Thu Sep 18 20:11:29 2014 Failed to stop KVM daemon in node '93eef5d9-6b33-4ce6-8f88-5353a093cb73': Node is marked offline
Modified node c1
 - master_candidate -> False
 - offline -> True

I then attempt a "gnt-node evacuate -p c1":

root@c0:~/gnt-node-evacuate# gnt-node evacuate -p c1
Relocate instance(s) wordpressx from node(s) c1?
y/[n]/?: y
Thu Sep 18 20:13:09 2014  - INFO: Evacuating instances from node 'c1.ws.nsrc.org': wordpressx
Thu Sep 18 20:13:09 2014  - WARNING: Unable to evacuate instances wordpressx (FailDisk)
Failure: command execution error:
Unable to evacuate instances wordpressx (FailDisk)

<b>What is the expected output? What do you see instead?</b>

Was expecting to be told that the node cannot be evacuated because it's down.

Arguably, I should be using gnt-node failover, but the response (FailDisk) leads me to believe something
is not right.



```
Originally added on 2014-09-20 10:24:15 +0000 UTC.

Attachments: 
[gnt-node-evacuate.tgz](https://storage.googleapis.com/google-code-attachments/ganeti/issue-951/comment-0/gnt-node-evacuate.tgz)

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-05-20 12:43:46 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
on master, the following works

- primary node of an instance goes offline
- gnt-node modify -O yes c1
- hbal -L -X

So, at least on master, there is a way to deal with this situation.
```
Originally added on 2015-05-28 18:33:12 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
related: issue 1096
```
Originally added on 2015-05-28 19:02:26 +0000 UTC.

Mentioned Issue with GoogleCode ID 1096 has been migrated to GitHub with ID #1152.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 4ca31587850b8d60b12f3be379cc0186387f2895
Author: Klaus Aehlig <aehlig@google.com>
Date:   Fri May 29 17:50:43 2015 +0200

    Allow unconditional failovers off offline nodes
    
    Normally, we should not place instances on nodes that do
    not have enough disks. However, there is one exception: if
    we failover an instance from an offline node, that node can
    well be secondary of that instance---the fact that it is currently
    primary proves it has enough disks space. The reason why we have
    to handle that case special is that if a node is offline, we sometimes
    cannot determine the amount of disk available, hence the conservative
    estimation is 0.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2015-05-29 16:20:26 +0000 UTC.

Added to Milestone: Release2.12

**Comment by Ganeti-Issues-Migrator** [comment]
```
I believe the mentioned patch fixes it; so it should all be fixed
in the next 2.12 release, i.e., 2.12.5.

Feel free to reopen, if that change needs to be ported down to 2.11.
```
Originally added on 2015-05-29 16:21:43 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:38:33 +0000 UTC.

Changed State: Released
